src/main/java/rx/internal/schedulers/NewThreadWorker.java

```diff
@@ -112,7 +124,7 @@ static void purgeExecutors() {
      * @return true if the policy was successfully enabled 
      */
     public static boolean tryEnableCancelPolicy(ScheduledExecutorService exec) {
-        if (!PURGE_FORCE) {
+        if (!PURGE_FORCE && (!IS_ANDROID || IS_ANDROID_AND_SUPPORTS_SET_REMOVE_ON_CANCEL_POLICY)) {
```

Comment:
@JakeWharton done! Take a look at `SHOULD_TRY_ENABLE_CANCEL_POLICY`.

@akarnokd done! Rewritten your code, added some tests and checked that it handles inheritance for `ScheduledExecutorService` (I was afraid that cached method won't work for child classes).

And tested again on JVM (Java 8), Android API 19 and Android API 22.

I should say that I noticed performance improvement in production Android app after these change for Android API < 21, hope it'll be in RxJava 1.0.14.